src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    const modifiedSeparator = `${separator}`;
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && this[i] !== undefined) {
+        result += this[i];
+        break;
+      } else if (i === this.length - 1 && this[i] === undefined) {
+        result += '';
+        break;
+      }
+
+      if (this[i] !== null && this[i] !== undefined) {
+        result += this[i] + modifiedSeparator;
+      } else {
+        result += modifiedSeparator;
+      }
```

Comment:
these `if` statements are redundant here. you can simply add `this[i]`(don't forget to check if `this[i] !== undefined, null` to convert it to an empty string if needed) to the result and `separator` if `i !== 0 || i !== this.length - 1`